File: sec3/p035.py
n = int(input())
l = list(map(int, input().split()))
cnt = 0
while all([i%2==0 for i in l]):
    cnt += 1
    l = [i//2 for i in l]
print(cnt)


print('#################################')

from collections import defaultdict
import math
n = int(input())
l = list(map(int, input().split()))
d = defaultdict(int)
for i in l:
    cnt = 0
    tmp = i
    while tmp % 2 == 0:
        cnt += 1
        tmp /= 2
    d[i] = cnt
print(d[min(d.keys())])

File: sec3/p035.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var scan = bufio.NewScanner(os.Stdin)

const (
	initBufSize = 10000
	maxBufSize = 10000000
	INF = (1 << 63) - 1
)

func init() {
	buf := make([]byte, initBufSize)
	scan.Buffer(buf, maxBufSize)
	scan.Split(bufio.ScanWords)
}

func main() {
	n := NumStdin()

	ans := INF
	for i := 0; i < n; i++ {
		x := NumStdin()

		cnt := 0
		for x % 2 == 0 {
			cnt++
			x /= 2
		}
		ans = min(ans, cnt)
	}

	fmt.Println(ans)
}

func NumStdin() int {
	scan.Scan()
	num, err := strconv.Atoi(scan.Text())
	if err != nil {
		panic(err)
	}
	return num
}

func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}
